src/main/java/seedu/address/logic/parser/EditAttendanceCommandParser.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CLASS_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LESSON_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PARTICIPATION_SCORE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STUDENT_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_WEEK;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.EditAttendanceCommand;
+import seedu.address.logic.commands.EditAttendanceCommand.EditAttendanceDescriptor;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.attendance.Week;
+
+/**
+ * Parses input arguments and create a new EditAttendanceCommand object.
+ */
+public class EditAttendanceCommandParser implements Parser<EditAttendanceCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditAttendanceCommand
+     * and returns an EditAttendanceCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public EditAttendanceCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_CLASS_INDEX, PREFIX_LESSON_INDEX,
+                PREFIX_STUDENT_INDEX, PREFIX_WEEK, PREFIX_PARTICIPATION_SCORE);
+
+        Index moduleClassIndex;
+        Index lessonIndex;
+        Index studentIndex;
+        Week week;
+
+        boolean isModuleClassIndexPresent = argMultimap.getValue(PREFIX_CLASS_INDEX).isPresent();
+        boolean isLessonIndexPresent = argMultimap.getValue(PREFIX_LESSON_INDEX).isPresent();
+        boolean isStudentPresent = argMultimap.getValue(PREFIX_STUDENT_INDEX).isPresent();
+        boolean isWeekPresent = argMultimap.getValue(PREFIX_WEEK).isPresent();
+        if (!isModuleClassIndexPresent || !isLessonIndexPresent || !isStudentPresent || !isWeekPresent) {
```

Comment:
Separate guard clause

```suggestion

        if (!isModuleClassIndexPresent || !isLessonIndexPresent || !isStudentPresent || !isWeekPresent) {
```

Reply:
@junlong4321 Actually, for such a case where the Boolean variables are only used in the guard clause, shouldn't they be part of the block for the guard clause?